### PR TITLE
[UIE-74] Type useStore hook

### PR DESCRIPTION
**Pull request by nawatts**

### CODE REVIEW

File: src/libs/react-utils.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import _ from 'lodash/fp'
...
+import { Atom, delay, pollWithCancellation } from 'src/libs/utils'
 
 
 /**
  * Performs the given effect, but only on component mount.
  * React's hooks eslint plugin flags [] because it's a common mistake. However, sometimes this is
  * exactly the right thing to do. This function makes the intention clear and avoids the lint error.
  */
-export const useOnMount = (fn: EffectCallback) => {
+export const useOnMount = (fn: EffectCallback): void => {
```
Comment (nawatts) [inline src/libs/react-utils.ts:13]: Unrelated. Clarifying type signatures on a few other hooks.

File: src/components/SupportRequest.js [hunk 2]
```diff
@@ -233,7 +233,7 @@ const SupportRequest = () => {
 const SupportRequestWrapper = () => {
   const isActive = useStore(contactUsActive)
 
-  return isActive && h(SupportRequest)
+  return isActive ? h(SupportRequest) : null
```
Comment (nawatts) [inline src/components/SupportRequest.js:236]: Components are intended to return a ReactElement or null.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e6c770ac6c25dcc7590f22e50a3f7dc582887da3/types/react/index.d.ts#L520-L526

File: src/libs/react-utils.ts [hunk 3]
```diff
@@ -144,11 +144,11 @@ export const useCurrentTime = (initialDelay = 250) => {
 /**
  * Hook that returns the value of a given store. When the store changes, the component will re-render
  */
-export const useStore = theStore => {
+export const useStore = <T>(theStore: Atom<T>): T => {
   const [value, setValue] = useState(theStore.get())
   useEffect(() => {
     return theStore.subscribe(v => setValue(v)).unsubscribe
-  }, [theStore, setValue])
+  }, [theStore])
```
Comment (nawatts) [inline src/libs/react-utils.ts:151]: `setState` is stable across renders and doesn't need to be included in `useEffect` dependencies.

https://reactjs.org/docs/hooks-reference.html#usestate

**Review by cahrens — APPROVED** [review]

**Review by msilva-broad — APPROVED** [review]
looks good

**Comment by nawatts** [comment]
Ignoring failed e2e test since this is a type change.

**Pull request merged**